### PR TITLE
Don't import from enterprise in OSS web client codebase

### PR DESCRIPTION
**Pull request by eseliger**
Seems like we've missed some cases of imports from enterprise in the non-enterprise package.
To prevent this from happening again, I also added a small script to check in CI.

This also has the positive side-effect that things that aren't available in the OSS version won't show in the site-admin UI navigation anymore. They were dead links that linked to a 404 page anyways. (Like the extensions link, codeintel,..)

### CODE REVIEW

File: client/web/src/contributions.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { ExtensionsControllerProps } from '../../shared/src/extensions/controlle
 import { registerHighlightContributions } from '../../shared/src/highlight/contributions'
 import { registerHoverContributions } from '../../shared/src/hover/actions'
 import { PlatformContextProps } from '../../shared/src/platform/context'
-import { registerSearchStatsContributions } from './enterprise/search/stats/contributions'
+import { registerSearchStatsContributions } from './search/stats/contributions'
```
Comment (eseliger) [inline client/web/src/contributions.ts:8]: Is this correct? It was always registered, but lives in enterprise.

File: dev/check/ts-enterprise-import.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+set -e
+
+if [ -n "$IMPORT_MATCHES" ]; then
+  echo
+  echo "Error: Found imports from enterprise codebase in client/web/src/(!enterprise)"
+  # shellcheck disable=SC2001
+  echo "$IMPORT_MATCHES" | sed 's/^/  /'
+
+  cat <<EOF
+Importing from enterprise in non-enterprise modules is forbidden.
```
Comment (valerybugakov) [inline dev/check/ts-enterprise-import.sh:24]: A quick search across the codebase doesn't give an answer on _why_ do we enforce this rule.
Would be great to link some markdown file where the reason is described.

**Review by valerybugakov — APPROVED** [review]

File: dev/check/ts-enterprise-import.sh (hunk 2 above)
Comment (eseliger) [reply dev/check/ts-enterprise-import.sh]: Added a link and short explanation to the error message!

File: client/web/src/contributions.ts (hunk 1 above)
Comment (eseliger) [reply client/web/src/contributions.ts:8]: This doesn't change any behavior, so I'm going to merge it as-is for now and we can always move it back to enterprise.

**Pull request merged**